src/chm/documents/customactions/wixsettingchange.html.md

```diff
@@ -0,0 +1,37 @@
+---
+title: WixBroadcastSettingChange and WixBroadcastEnvironmentChange Custom Actions
+layout: documentation
+after: using_standard_customactions
+---
+
+# WixBroadcastSettingChange and WixBroadcastEnvironmentChange Custom Actions
+
+The WixBroadcastSettingChange and WixBroadcastEnvironmentChange custom actions are immediate custom actions that send a WM\_SETTINGCHANGE message to all top-level windows indicating that settings have changed. WixBroadcastSettingChange indicates that unspecified settings have changed. WixBroadcastEnvironmentChange indicates that environment variables have changed.
+
+Other programs are supposed to listen for WM\_SETTINGCHANGE and update any internal state with the new setting.
+
+Windows Installer itself sends the WM\_SETTINGCHANGE message for settings it changes while processing an MSI package but cannot do so for changes a package makes via custom action. We've also received reports that Windows Installer does not reliably send WM\_SETTINGCHANGE for environment variable changes.
```

Comment:
"We've also received reports..." sounds like we're making conjectures. It's actually documented why on my blog and confirmed from the Installer team (Carolyn, IIRC): WM_SETTINGCHANGE is not sent when a reboot is pending (whether from the current MSI or before it).